SingularityBase/src/main/java/com/hubspot/singularity/SingularityRequestHistory.java

```diff
@@ -93,6 +72,27 @@ public long getCreateTimestampForCalculatingHistoryAge() {
   }
 
   @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    SingularityRequestHistory that = (SingularityRequestHistory) o;
+    return createdAt == that.createdAt &&
+        Objects.equals(user, that.user) &&
+        eventType == that.eventType &&
+        Objects.equals(request, that.request) &&
+        Objects.equals(message, that.message);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(createdAt, user, eventType, request, message);
+  }
+
+  @Override
   public String toString() {
     return "SingularityRequestHistory [createdAt=" + createdAt + ", user=" + user + ", eventType=" + eventType + ", request=" + request + ", message=" + message + "]";
   }
```

Comment:
nit: should we switch this to: 
```
"SingularityRequestHistory{" +
"createdAt=" + createdAt + 
", user=" + user + 
", eventType=" + eventType + 
", request=" + request+ 
", message=" + message + 
"}";
```
to keep the format consistent with the others? Looks like there are a few other places using `[]` and formatted differently